src/mono/mono/metadata/icall.c

```diff
@@ -4106,9 +4106,7 @@ ves_icall_RuntimeType_GetPropertiesByName_native (MonoQCallTypeHandle type_handl
 		g_hash_table_insert (properties, prop, prop);
 	}
 	if (!(bflags & BFLAGS_DeclaredOnly) && (klass = m_class_get_parent (klass))) {
-		// BFLAGS_NonPublic should be excluded for base classes
-		bflags &= ~BFLAGS_NonPublic;
```

Comment:
The rationale for excluding this flag on base classes previously, does it no longer apply anymore?
Does the test `System.Reflection.Tests.TypeTests_HiddenTestingOrder.HideDetectionHappensAfterPrivateInBaseClassChecks` still pass then? `BFLAGS_NonPublic` was being unset here https://github.com/mono/mono/pull/14449 to fix that test.

Reply:
It was passing locally, waiting for CI to see.

Reply:
I dont think it will run on CI for Android, it seems like only runtime tests are ran right now. https://github.com/dotnet/runtime/pull/77280 is looking to change that.

Reply:
Looks like the test suite System.Reflection.Tests is disabled on Android on CI right now because of crashes in the past. 

I think if we remove this, it would make this condition https://github.com/dotnet/runtime/blob/bbcff6bc422da3d9b1904281a86014236ce50f2e/src/mono/mono/metadata/icall.c#L4075 always true (since we will never make the bit 0 and it will be set to 1 in the beginning of the function). So we might be able to clean up some things like remove https://github.com/dotnet/runtime/blob/1c265ec2e006c569e538792a65bcef77146af8e6/src/mono/mono/metadata/icall.c#L4035-L4036 and collapsing https://github.com/dotnet/runtime/blob/1c265ec2e006c569e538792a65bcef77146af8e6/src/mono/mono/metadata/icall.c#L4075-L4076 into just 
```
else if (property_accessor_nonpublic(prop->get, startklass == klass) ||
				property_accessor_nonpublic(prop->set, startklass == klass)) {
				match++;
			} 
```

From what I've seen, I'm not actually sure why "`BFLAGS_NonPublic` should be excluded for base classes" was the fix for `HideDetectionHappensAfterPrivateInBaseClassChecks` as the current test does not have base classes with non public properties.

Reply:
I'm also not sure why we need to exclude non public properties for base class as initial issue was that private properties in subclasses can hide public properties with the same name in Parents, and bflags &= ~BFLAGS_NonPublic will hide ALL non-public base class props.